src/styles/getModalStyles/getModalStyles.js

```diff
@@ -1,9 +1,9 @@
-import CONST from '../CONST';
-import colors from './colors';
-import variables from './variables';
-import themeColors from './themes/default';
+import CONST from '../../CONST';
+import colors from '../colors';
+import variables from '../variables';
+import themeColors from '../themes/default';
 
-export default (type, windowDimensions, popoverAnchorPosition = {}, containerStyle = {}) => {
+export default ({shouldModalAddTopSafeAreaPadding = {}}) => (type, windowDimensions, popoverAnchorPosition = {}, containerStyle = {}) => {
```

Comment:
Here I modified the current function with a new function that accept parameters that can be determined what modal type should apply the top padding.

The previous code has `shouldAddTopSafeAreaPadding` returning `true` on `CONST.MODAL.MODAL_TYPE.CENTERED`, `CONST.MODAL.MODAL_TYPE.CENTERED_UNSWIPEABLE`, and `CONST.MODAL.MODAL_TYPE.RIGHT_DOCKED`. For android we can specified `CONST.MODAL.MODAL_TYPE.CENTERED` and `CONST.MODAL.MODAL_TYPE.CENTERED_UNSWIPEABLE` no to add top padding by returning `shouldAddTopSafeAreaPadding` with `false`.

Reply:
@mollfpr 

I think `shouldAddSafeAreaPadding` could be a library where computations can happen in platform-specific files.
Could be added inside the getModalStyle folder itself. 

Any thoughts @AndrewGable?

Reply:
@Santhosh-Sellavel Can you outline your suggested fix?

Reply:
@Santhosh-Sellavel Can you outline your suggested fix?

Reply:
@AndrewGable 

## Suggestion 1
Could be as simple as the library name `shouldAddSafeAreaPadding` where `index.ios.js` returns true and `index.js` returns `false`. And call it from the appropriate line.

## Suggestion 2
Could be as simple as the library name `shouldAddSafeAreaPadding` where 
`index.ios.js`
`index.android.js`
`index.js`

return appropriate values based on MODAL_TYPE.

Reply:
I think suggestion 2 will be good solution.